### DISCUSSION TITLE
Some performance data

### DISCUSSION THREAD

**Discussion by dibyendumajumdar** [dibyendumajumdar/ravi#235] [Show and tell]
Some numbers obtained from running ravi-simplex over netlib data.

* Lua 5.4.3 [lua-results.txt](https://github.com/dibyendumajumdar/ravi/files/6913444/lua-results.txt)
* Ravi interpreted [ravi-nojit-results.txt](https://github.com/dibyendumajumdar/ravi/files/6913447/ravi-nojit-results.txt)
* Ravi JIT (from Lua bytecodes) [ravi-oldjit-results.txt](https://github.com/dibyendumajumdar/ravi/files/6913448/ravi-oldjit-results.txt)
* Ravi JIT (from source) [ravi-newjit-results.txt](https://github.com/dibyendumajumdar/ravi/files/6913446/ravi-newjit-results.txt)
* Ravi AOT [ravi-aot-results.txt](https://github.com/dibyendumajumdar/ravi/files/6913445/ravi-aot-results.txt)

**Comment by lufeidefei** [comment]
i have tested the performace use https://github.com/gligneul/Lua-Benchmarks。（i download the code and compile by myself）。here is the result. maybe i miss something.
test    lua-5.4.3   lua-5.3.5   ravi    ravi-jit    
ack 1.239   1.8 1.777   2.246   
fixpoint-fact   1.874   1.995   1.697   1.736   
heapsort    2.906   4.712   3.242   3.017   
mandelbrot  2.649   2.799   2.854   2.97        
juliaset    2.15    2.802   2.498   2.456   
queen   2.981   4.085   3.043   3.111   
sieve   1.715   2.741   2.248   2.236   
binary  2.879   7.665   5.852   6.711   
n-body  4.914   5.921   5.668   5.291   
fannkuch    3.906   7.004   4.701   4.314   
fasta   3.513   4.831   4.083   4.156   
k-nucleotide    0.003   0.002   0.003   0.003   
spectral-norm   4.062   6.061   4.882   5.3

**Reply by dibyendumajumdar** [reply]
hi @lufeidefei 

You have will to modify the scripts for Ravi to get best performance. Ravi requires type annotations; code that use numeric arrays and numeric processing will show the best results.

There is no benefit running standard Lua programs in Ravi as performance of JIT is better only when there are type annotations.

Regards

**Reply by lufeidefei** [reply]
Thanks for reply! Got it. I will try.
